main.py

```diff
@@ -117,7 +118,7 @@ def sample_and_publish_task(scope, delay):
             except KeyError:
                 continue
             logging.info("publishing %s %s", name, value)
-            plugin.publish(name, value=value, scope=scope, timestamp=timestamp)
+            plugin.publish(name, value=value, scope=scope, meta={"zone": args.host}, timestamp=timestamp)
```

Comment:
I would think that posting the `zone` that something is published in would be the job of the `plugin.publish` function and not the responsibility of the plugin.

With the development model you have here, we are requiring that `--host` be an argument to every plugin that needs to publish it's zone.

a) the plugin shouldn't care what `zone` it is running in and therefore shouldn't need that knowledge
b) the `zone` is similar to the `host` or `vsn` or `node` and should be added to the `meta` by `plugin.publish` automatically.

Reply:
My impression on this was that the plugin would decide whether they want to publish the "zone" attribute or not because this would be an unnecessary addition to object counter which does not care what zone it was run on. But, as long as end users understand what the zone means, it may be fine to include that attribute to all measurements we publish.